src/module.c

```diff
@@ -50,7 +50,7 @@ static int parseLabelsFromArgs(RedisModuleString **argv, int argc, size_t *label
         	RedisModule_StringPtrLen(key, &keyLen);
         	RedisModule_StringPtrLen(value, &valueLen);
         	if(keyLen==0 || valueLen==0 || strpbrk( RedisModule_StringPtrLen(value, NULL), "(),")){
-        		free(labelsResult);
+        		FreeLabels(labelsResult, i); // need to release prior key values too
```

Comment:
```suggestion
        		FreeLabels(labelsResult, i + 1); // need to release prior key values too
```

Reply:
why i+1?
The current round wasn't assigned yet
e.g. on the first round when i=0 I don't have anything to release